packages/amplify-category-function/src/provider-utils/awscloudformation/index.ts

```diff
@@ -343,13 +350,22 @@ export async function updateConfigOnEnvInit(context: $TSContext, resourceName: s
       envParams = await initTriggerEnvs(context, resourceParams, providerPlugin, envParams, srvcMetaData);
     }
 
+    if (resourceParams.lambdaLayers.length) {
+      const envName = context.amplify.getEnvInfo().envName;
+      const modifiedLambdaLayers: LambdaLayer[] = [];
+      modifiedLambdaLayers.push(...convertProjectLayersToExternalLayers(resourceParams.lambdaLayers, envName));
+      modifiedLambdaLayers.push(...covertExternalLayersToProjectLayers(resourceParams.lambdaLayers, envName));
+      resourceParams.lambdaLayers = modifiedLambdaLayers;
+      JSONUtilities.writeJson(functionParametersPath, resourceParams);
+    }
+
     return envParams;
   } else if (isMultiEnvLayer(resourceName) && service === ServiceName.LambdaLayer) {
     const projectPath = pathManager.findProjectRoot();
     const currentAmplifyMeta = stateManager.getCurrentMeta(projectPath);
     const amplifyMeta = stateManager.getMeta(projectPath);
     const currentCloudVersionHash: string = _.get(currentAmplifyMeta, [categoryName, resourceName, 'versionHash'], undefined);
-
+    //TODO: check if we require version hash when changing environment
```

Comment:
We need to compare the contents of the layer folder to the last known layer version hash of the environment in order to correctly whether there are changes to the newly checked out environment. This is needed for `amplify status` and deploying the correct layer versions on push.